fs-util.js

```diff
@@ -0,0 +1,32 @@
+var fs = require('fs');
+var path = require('path');
+
+var IMPORT_CACHE = {};
+
+/**
+ * Checks for file path for existence and save the result in cache to speed up the next check for the same path.
+ * USE CASE: prevent fs calls for the same folder when importing various sass files from a single module (e.g bootstrap)
+ * - module/label.scss
+ * - module/button.scss
+ * @param {String} filePath - path to check for existence
+ */
+function existsSync(filePath) {
+  if (!(filePath in IMPORT_CACHE)) {
+    IMPORT_CACHE[filePath] = fs.existsSync(filePath);
+  }
+
+  return IMPORT_CACHE[filePath];
+}
+
+function isDirectoryImport(filePath) {
+  return !path.extname(filePath) && existsSync(filePath);
+}
+
+function isFileImport(filePath) {
+  return existsSync(path.dirname(filePath));
+}
```

Comment:
I am a little confused by the naming of these functions. The names imply that they check if there is an extension or not but under the hood they actually perform an `fs.existsSync` check. Also the `isFileImport` function checks if the dirname of the filepath exists which doesn't quite perform the task that the name suggests i.e. i could have `my-dir-name/foo.scss` which would return true if `my-dir-name` exists but `foo.scss` didn't.

Reply:
yeah I agree with you it's not very obvious and does not do exactly what the name implies. The problem is that we usually don't use file extensions when we import things in sass. Which mean we can not check the whole import path for existence, since it wouldn't exist anyway for file imports (e.g `my-package/button`). Previously we checked the directory name for the import, which is the `isFileImport` method here, I ported that as it is. I called it file import since we supported only file import so far. The other thing is directory import when the full import path exist, like `my-package`. If that thing exists then we treat that like a directory import rather than a file import, that's why I called it `isDirectoryImport`. Do you have any name suggestion?

Reply:
I'd:
1. get rid of `isDirectoryImport` and replace it's usage with `!path.extname(filePath) && existsSync(filePath)`. It's short enough that it doesn't warrant a wrapper function. If you want, you could wrap the `!path.extname(filePath)` in a function called `isDirectory` to make it clearer.
2. get rid of `isFileImport` and just use `existsSync(path.dirname(filePath))`
3. Revisit why `existsSync(path.dirname(filePath))` doesn't just check the full path i.e `existsSync(filePath)`